app/components/option_item/option_icon.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useMemo, useState} from 'react';
+import FastImage from 'react-native-fast-image';
+
+import CompassIcon from '@components/compass_icon';
+import {useTheme} from '@context/theme';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {isValidUrl} from '@utils/url';
+
+type OptionIconProps = {
+    destructive?: boolean;
+    icon: string;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        destructive: {
+            color: theme.dndIndicator,
+        },
+        icon: {
+            fontSize: 24,
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            width: 24,
+            height: 24,
+        },
+    };
+});
+
+const OptionIcon = ({icon, destructive}: OptionIconProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const imageStyle = useMemo(() => (
+        [styles.icon, destructive ? styles.destructive : null]
+    ), [styles, destructive]);
+
+    const [failedToLoadImage, setFailedToLoadImage] = useState(false);
+    const onErrorLoadingIcon = useCallback(() => {
+        setFailedToLoadImage(true);
+    }, []);
+
+    const iconAsSource = useMemo(() => {
+        return {uri: icon};
+    }, [icon]);
+
+    if (isValidUrl(icon) && !failedToLoadImage) {
+        return (
+            <FastImage
+                source={iconAsSource}
+                style={imageStyle}
```

Comment:
This styles have some colors in them, but to tint the image a tintColor prop is expected, is that being taken into consideration?

Reply:
@enahum Note that will only render here if this is a URL image. Not a compass image, and also likely not an svg. Should we remove everything "color" about the styling of this `FastImage` element here?

Reply:
I don't think the intended plan is to tint the images at all, in order to keep the branding of the given integration. Do you suggest we tint the color anyway?